std/socket.d

```diff
@@ -1993,7 +2012,9 @@ static if (is(sockaddr_un))
 
         @property string path() @trusted const pure
         {
-            return to!string(cast(const(char)*)sun.sun_path.ptr);
+            return (cast(const(char)*) sun.sun_path.ptr)[0 ..
+                _nameLen - sockaddr_un.init.sun_path.offsetof -
+                (sun.sun_path.ptr[0] ? 1 : 0)].idup;
         }
```

Comment:
Why exclude the null byte from the result? It was present in the constructor, so excluding it here would be inconsistent. Also, without it, it seems like it's no longer possible to know whether a particular `UnixAddress` represents an abstract path or not.